docs/dev/Lua API.rst

```diff
@@ -5766,82 +5766,129 @@ The names of the functions are also available as the keys of the
 
-    :frameLength:
-        how many ticks does one frame take OR
-    :isMechanical:
-        a bool that says to try to match to mechanical system (i.e. how gears are turning)
+  Setup and enable machine-like functionality for the workshop. Note: due to implementation limitations
+  workshop only connects to other machines if the other machine is build later than this one.
 
-``getPower(building)`` returns two number - produced and consumed power if building can be modified and returns nothing otherwise
+  :workshop_type:   custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :needs_power:   only function if it has sufficient power
+  :power_consumed:   building consumes this amount of power
+  :power_produced:   output this amount of power
+  :connection_points:   a table of ``{x=?,y=?}`` points that can connect to other machines
 
-``setPower(building,produced,consumed)`` sets current power production and consumption for a building.
+* ``setMachineInfoAuto(workshop_type, needs_power, power_consumed, power_produced, gear_tiles)``
+
+  Same as ``setMachineInfo`` but fills out the ``connection_points`` table from raws placing connection
+  points on tiles which have the gear tile. ``gear_tiles`` is an optional array of two tiles that are
+  counted as gears in the workshop ascii tile raws.
+
+* ``setAnimationInfo(workshop_type, frames, frame_skip)``
+
+  Animate workshop by replacing displayed tiles (or graphical tiles). There are two ways this works:
+  if ``frame_skip>=0`` then it shows each frame for ``frame_skip`` of frames or if ``frame_skip<0``
+  Frames are synchronized to the machine this building is connected to.
+
+  :workshop_type:   custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :frames:          table of frames. Each frame is sparse flat table with ids from ``0`` to ``31*31-1``.
+                    Each frame tile is table of integers from 4 to 8 members long. Tile members are as
+                    follow: tile, foreground color, background color, bright, graphics tile, overlay tile,
+                    signpost tile, item tile.
+  :frame_skip:      How many ticks to display one frame. If set to negative number (or skipped) frames
+                    are synchronized with machine animation.
+
+* ``setAnimationInfoAuto(name, make_graphics_too, frame_skip, gear_tiles)``
+
+  Animate workshop as with function above but generate frames automatically. This works by finding
+  tiles which have gears and animating them with alternating gear tiles.
+
+  :name:            custom workshop string id e.g. ``SOAPMAKER``
+  :make_graphics_too:  replace same tiles in graphics mode with tiles from vanilla df mechanism
+  :frame_skip:      How many ticks to display one frame. If set to negative number (or skipped) frames
+                    are synchronized with machine animation.
+  :gear_tiles:      Optional array of 2 or 4 indexes. First two define ascii tiles and next two graphics tiles
```

Comment:
is this for overriding the default gear graphics? If so, that should be communicated.